src/diffusers/models/attention_processor.py

```diff
@@ -509,6 +510,15 @@ def forward(
         # The `Attention` class can call different attention processors / attention functions
         # here we simply pass along all tensors to the selected processor class
         # For standard processors that are defined here, `**cross_attention_kwargs` is empty
+
+        attn_parameters = set(inspect.signature(self.processor.__call__).parameters.keys())
+        unused_kwargs = [k for k, _ in cross_attention_kwargs.items() if k not in attn_parameters]
+        if len(unused_kwargs) > 0:
+            logger.warning(
+                f"cross_attention_kwargs {unused_kwargs} are not expected by {self.processor.__class__.__name__} and will be ignored."
+            )
+        cross_attention_kwargs = {k: w for k, w in cross_attention_kwargs.items() if k in attn_parameters}
```

Comment:
Could I get an explanation on how this ensures the `scale` parameter from `cross_attention_kwargs` gets passed appropriately?

Reply:
It won't get passed if the attention processor does not accept a `scale` parameter - it will throw a warning instead. 

For example, `IPAdapterAttnProcessor` has a `scale` parameter in its signature, but it does not use it. We only added it so that we can use it together with the default attention processors 
https://github.com/huggingface/diffusers/blob/f07899a57c8ac8e992a910dc203ff9ee46f363b5/src/diffusers/models/attention_processor.py#L2133

With this, we don't need to do that anymore. - Attention Processors will have fixed signatures that only include parameters that they actually use

Reply:
another example is in this PR https://github.com/huggingface/diffusers/pull/6847/files#diff-96aeb208ac8e69bdb368e0a481a0cd50173b3f4bf19e5da500c251614820c8cdR1201

it introduced a new parameter for IPAdapterAttnProcessor `ip_adapter_mask` - and they had to add same parameter to AttenProcessor.__call__ in order for these two attention processors to work together. We don't need to do that too

Reply:
Thanks for explaining. The PR is now a go.